### PR TITLE
Extract department specific text to locale file

### PR DESCRIPTION
**Pull request by dwhenry**
**NOTE:** This pull request is an example only and not fully implemented.

This is a proposed solution that would allow the PensionWise team to use the Signonotron code with different text at key points within the application without requiring multiple files to be changed within a branch.

This would make merging changes from the main application easy as only the locale file would be significantly different.

### CODE REVIEW

**Comment by boffbowsh** [comment]
It might be better to follow [12 factor config](http://12factor.net/config) principles here and allow the Issuer (and other variables) to be overridden in the environment. Then the change is a deploy-time concern.

**Comment by dwhenry** [comment]
Hi Paul,

Total agree, apart from the size, would estimate about 50-60 bit of text
that could be different, this size could be significantly reduces depending
on how words are grouped. I still feel that using environment variables
would become cumbersome, but happy to investigate further and give
actual numbers - as mine are just gut feel.

David

**Comment by boffbowsh** [comment]
Fair point. So I guess you'd convert everything to I18n format and then replace the locale file at deploy time? If so, :+1: for the approach

**Comment by dwhenry** [comment]
@boffbowsh @benlovell I'm not happy with the view changes (e534dd) I have made.

I'm going to create a separate pull request without any view changes and then look to refactor the mailer views to use a layout - reducing duplication - which will hopefully put me in a better place for the view changes I want to make. 

Thoughts?

**Comment by boffbowsh** [comment]
Sounds ok to me :+1:

File: app/controllers/devise/two_step_verification_controller.rb [hunk 1]
```diff
@@ -27,7 +27,7 @@ def update
   end
 
   def otp_secret_key_uri
-    issuer = "GOV.UK%20Signon"
+    issuer = I18n.t('devise.issuer')
```
Comment (jamiecobbett) [inline app/controllers/devise/two_step_verification_controller.rb]: Won't this need to be URI encoded?

Comment (dwhenry) [inline app/controllers/devise/two_step_verification_controller.rb]: @jamiecobbett  you are correct - had done that in the locale file. This has been fixed in a separate branch https://github.com/alphagov/signonotron2/pull/471

**Pull request closed without merging**